gruntwork-install

```diff
@@ -0,0 +1,177 @@
+  local readonly branch="$3"
+  local readonly github_token="$4"
+  local readonly download_path="$5"
+
+  # We want to make sure that all folders down to $download_path/$module_name exists, but that $download_path/$module_name itself is empty.
+  mkdir -p "$download_path/$module_name/"
+  rm -Rf "$download_path/$module_name/"
+
+  # Note that fetch can safely handle blank arguments for --tag or --branch
+  # If both --tag and --branch are specified, --branch will be used
+  # TODO: fetch should read GITHUB_OAUTH_TOKEN as an environment variable too
+  fetch --repo="$SCRIPT_MODULES_REPO" --tag="$tag" --branch="$branch" --github-oauth-token="$github_token" "/modules/$module_name" "$download_path/$module_name" >/dev/null
+}
+
+# Validate that at least one file was downloaded from the module; otherwise throw an error.
+function validate_module {
+  local readonly module_name="$1"
+  local readonly download_path="$2"
+
+  if [[ ! -e "$download_path/$module_name" ]]; then
+    echo "ERROR: No files were downloaded. Are you sure \"$module_name\" is a valid Script Module in $SCRIPT_MODULES_REPO?"
+    exit 1
+  fi
+}
+
+# Take in an array of key-val pairs in this format:
+#   module_params[0] = "key1=value1"
+#   module_params[1] = "key2=value2"
+# .. and convert it to the format that Gruntwork bash scripts expect:
+#   --key1 value1 --key2 value2
+function convert_module_params_format {
+    local readonly module_params=("${@}")
+
+    module_params_formatted=""
+    for key_val_expression in "${module_params[@]}"; do
+       key="${key_val_expression%=*}"
+       val="${key_val_expression#*=}"
+       module_params_formatted="--${key} ${val} ${module_params_formatted}"
+    done
+
+    echo $module_params_formatted
+}
+
+function run_module {
+  local readonly module_name="$1"
+  shift
+  local readonly module_params=($@)
+  local readonly module_params_formatted=$(convert_module_params_format "${module_params[@]}")
+
+  chmod -R u+x "${MODULES_DOWNLOAD_DIR}/${module_name}"
+  eval "${MODULES_DOWNLOAD_DIR}/${module_name}/${MODULE_INSTALL_FILE_NAME} ${module_params_formatted}"
+}
+
+function install_script_module {
+  local tag=""
+  local branch=""
+  local module_name=""
+  local module_params=()
+
+  while [[ $# > 0 ]]; do
+    local key="$1"
+
+    case "$key" in
+      --tag)
+        tag="$2"
+        shift
+        ;;
+      --branch)
+        branch="$2"
+        shift
+        ;;
+      --module-name)
+        module_name="$2"
+        shift
+        ;;
+      --module-param)
+        module_param="$2"
+        module_params+=("$module_param")
+        shift
+        ;;
+      --help)
+        print_usage
+        exit
+        ;;
+      *)
+        echo "ERROR: Unrecognized option: $key"
+        print_usage
+        exit 1
+        ;;
+    esac
+
+    shift
+  done
+
+  assert_is_installed fetch
+  assert_not_empty "--module-name" "$module_name"
+  assert_env_var_not_empty "GITHUB_OAUTH_TOKEN"
+
+  echo "Installing $module_name..."
```

Comment:
These echo statements may get lost in a sea of output. Consider calling them out somehow. Minor issue.